processor/src/main/java/org/mapstruct/ap/internal/model/GeneratedType.java

```diff
@@ -169,6 +169,10 @@ public Accessibility getAccessibility() {
         return accessibility;
     }
 
+    public void setConstructor(Constructor constructor) {
+        this.constructor = constructor;
```

Comment:
We need to call `getImportTypes` on the `Constructor`. For that we also need to add the signature to the `Constructor` interface as well.